evil-collection.el

```diff
@@ -112,6 +114,62 @@ this confusing. It will be included if
   :type '(repeat (choice symbol sexp))
   :group 'evil-collection)
 
+;;;###autoload
+(cl-defun evil-collection-translate-key (state keymap-name
+                                               &rest maps
+                                               &key destructive
+                                               &allow-other-keys)
+  "Translate keys in the keymap corresponding to STATE and KEYMAP-NAME.
+STATE should be the name of an evil state or nil. KEYMAP-NAME should be a symbol
+corresponding to the keymap to make the translations in. MAPS corresponds to a
+list of translations (key replacement pairs). For example, specifying \"a\"
+\"b\" will bind \"a\" to \"b\"'s definition in the keymap.If DESTRUCTIVE is
+non-nil, the keymap will be destructively altered without a backup being
+created. If DESTRUCTIVE is nil, a backup of the keymap will be stored on the
+initial invocation, and future invocations will always look up keys in the
+backup keymap. On the other hand, if DESTRUCTIVE is non-nil, calling this
+function multiple times with \"a\" \"b\" \"b\" \"a\", for example, would
+continue to swap and unswap the definitions of these keys. This means that when
+DESTRUCTIVE is non-nil, all related swaps/cycles should be done in the same
+invocation."
+  (declare (indent defun))
+  (let* ((keymap (if state
```

Comment:
I might be wrong here, but taking a look at the state variable, we're constraining it to only passing in one 'state variable right?

If so, I'm curious if we want to support more states at once, kinda similar to evil-define-key.

(evil-define-key '(normal visual) keymap key command)

Reply:
I think supporting several states would be quite useful, although not mandatory.